### ISSUE TITLE
`missing-heading-in-file` after upgrade to latest `remark-cli` (v10)

### ISSUE THREAD

**Issue by travi** [remarkjs/remark-validate-links#62]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

remark-cli v10.0.0, remark v14.0.0, remark-lint v8.0.0,  remark-validate-links v10.0.4

### Link to runnable example

https://github.com/travi-test/lerna-playground/tree/remark-heading-in-file

### Steps to reproduce

1. change into the `packages/foo` directory
1. `npm install`
1. `npm run lint:md`

the failing link does point to a real heading in a real document in the same repo:
https://github.com/travi-test/lerna-playground/blob/2efe15d21124a94d525c26d92113bcb43495467b/packages/foo/README.md#L65

### Expected behavior

the heading in the referenced document exists, so it should not be an error. i added to this repo for reproduction purposes, but the private project where i first encountered this was previously not failing before the `remark-cli` upgrade to v10

### Actual behavior

i receive the following error:

```sh
$ npm run lint:md
> @travi/foo@0.0.0-semantically-released lint:md
> remark . --frail

README.md
  65:3-65:22  warning  Link to unknown heading in `../bar`: `usage`  missing-heading-in-file  remark-validate-links

⚠ 1 warning
```

### Runtime

Node v16

### Package manager

npm v7

### OS

macOS

**Comment by wooorm** [comment]
We’re in the process of updating everything. Expect a new release of this plugin in a couple of days!

**Comment by travi** [comment]
fair enough. i wondered if that would be the case with all of the recent activity. i'll keep my eye out. thanks!

**Comment by wooorm** [comment]
This was most likely solved in [v10](https://github.com/remarkjs/remark-validate-links/releases/tag/11.0.0)!

**Comment by travi** [comment]
it doesnt appear that this ended up being fixed. i upgraded to v10 in the branch mentioned above and still see the same problem: https://github.com/travi-test/lerna-playground/commit/2cd19f848fdc850204fec5c804db6509dcfd6718

**Comment by wooorm** [comment]
I’m not sure how to reproduce this problem with that repo at that commit. I do see that package locks include `remark-validate-links` at 10.0.4 instead of 11?

**Comment by wooorm** [comment]
btw, you mention remark-usage, but this issue is about remark-validate-links?

**Comment by travi** [comment]
ah shoot, you're right. i got confused with so many updates happening right now. looks like that particular project is using `remark-validate-links` through my preset. i'll try to work through getting that updated or verify in a more isolated scenario soon.

**Comment by wooorm** [comment]
I use a lib called 'npm-check-updates' locally, which helps me find what can be updated! Might be of help.

**Comment by travi** [comment]
the dependency dance isnt my issue. i heavily use renovate to stay up to date and use [updtr](https://npm.im/updtr) locally, but appreciate the tip either way. i've been considering more use of `npm-check-updates`, but `updtr` is still doing the job for me.

the bigger issue was just that i wasnt directly depending on `remark-validate-links` in the reproduction. i decided to go ahead and get my presets converted over to ESM-only and doing so confirmed that your fix did resolve the problem. i've still got a few ESM details to sort out in the pre-releases of my presets, but the latest commit in the branch i've been mentioning in this thread shows everything passing again.

thanks for the additional guidance and sorry for the extra noise.
